ebos/eclwriter.hh

```diff
@@ -372,9 +376,44 @@ private:
 
     Opm::NNC exportNncStructure_() const
     {
-        Opm::NNC nnc = eclState().getInputNNC();
-        int nx = eclState().getInputGrid().getNX();
-        int ny = eclState().getInputGrid().getNY();
+        std::size_t nx         = eclState().getInputGrid().getNX();
+        std::size_t ny         = eclState().getInputGrid().getNY();
+        auto nncData           = eclState().getInputNNC().nncdata();
+        auto editnncData       = eclState().getInputEDITNNC().data();
+        const auto& unitSystem = simulator_.vanguard().deck().getActiveUnitSystem();
+        auto edited = Detail::sortNncAndApplyEditnnc(nncData, editnncData);
+        std::vector<Opm::NNCdata> outputNnc;
+        std::size_t index = 0;
+
+        for( auto& entry : nncData )
```

Comment:
`const auto&`?

Reply:
Nope. I need to change the entry to faciliate searching (cell1<=cell2 needs to hold). See a few lines below.

Reply:
that's what I'd call surprising behavior: if some objects in a container needs to be modified, it should be much more easily recognizable, e.g. `container[nncIdx].value = foo();`...